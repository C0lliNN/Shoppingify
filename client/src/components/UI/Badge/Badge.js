import styled from 'styled-components';
import {
  COLORS,
  BORDER_RADIUS_1,
  FONT_SIZE_1,
} from '../../../helpers/style-constants';
import PropTypes from 'prop-types';

const Badge = styled.span`
  border: 1px solid ${(props) => COLORS[props.variant]};
  border-radius: ${BORDER_RADIUS_1};
  color: ${(props) => COLORS[props.variant]};
  padding: 3px 10px;
  font-weight: 500;
  font-size: ${FONT_SIZE_1};
`;

Badge.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
  variant: PropTypes.oneOf(Object.keys(COLORS)).isRequired,
};

export default Badge;
