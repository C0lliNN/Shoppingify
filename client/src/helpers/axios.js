import axios from 'axios';
import store from '../store/index';

export default function getAxios() {
  const config = {
    headers: {},
  };
  // eslint-disable-next-line no-undef
  if (process.env.NODE_ENV === 'development') {
    config.baseURL = 'http://localhost:5000/api/v1';
  } else {
    config.baseURL = '/api/v1';
  }

  config.headers['Content-Type'] = 'application/json';

  const token = store.getState().auth.token;

  if (token) {
    config.headers['Authorization'] = `Bearer ${token}`;
  }

  return axios.create(config);
}
