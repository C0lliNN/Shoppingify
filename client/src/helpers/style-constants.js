export const COLORS = {
  primary: '#F9A109',
  secondary: '#56CCF2',
  danger: '#EB5757',
  white: '#FFFFFF',
  gray_1: '#E0E0E0',
  gray_2: '#BDBDBD',
  gray_3: '#828282',
  black_1: '#000000',
  black_2: '#34333A',
  burgundy: '#80485b',
  light_brown: '#fff0de',
};

export const BUTTON_TYPES = ['raised', 'outlined', 'flat'];

export const FONT_FAMILY = 'Quicksand';
export const FONT_SIZE_1 = '12px';
export const FONT_SIZE_2 = '14px';
export const FONT_SIZE_3 = '18px';
export const FONT_SIZE_4 = '24px';

export const BORDER_RADIUS_1 = '12px;';
export const BORDER_RADIUS_2 = '24px;';

export const MD_BREAK_POINT = 768;
export const LG_BREAK_POINT = 1200;

export const NAVBAR_XS_SIZE = 60;
export const NAVBAR_MD_SIZE = 90;

export const INFO_BAR_MD_SIZE = 250;
export const INFO_BAR_LG_SIZE = 370;
