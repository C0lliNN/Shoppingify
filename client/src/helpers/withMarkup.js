function getByTextWithMarkup(getByText, text) {
  return getByText((content, node) => {
    const hasText = (node) => node.textContent === text;

    const nodeHasText = hasText(node);
    const childrenDontHaveText = Array.from(node.children).every(
      (child) => !hasText(child)
    );

    return nodeHasText && childrenDontHaveText;
  });
}

export default getByTextWithMarkup;
