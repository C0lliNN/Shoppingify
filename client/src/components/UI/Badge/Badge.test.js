import React from 'react';
import Badge from './Badge';
import { render, cleanup } from '@testing-library/react';
import { COLORS } from '../../../helpers/style-constants';
import 'jest-styled-components';

describe('<Badge/>', () => {
  let errorMessage = null;
  beforeEach(() => {
    console.error = jest.fn((message) => (errorMessage = message));
  });
  it('should console.error if no children is defined', () => {
    render(<Badge variant="primary"></Badge>);

    expect(console.error).toHaveBeenCalled();
    expect(errorMessage).toMatch(/.*children.*required.*/i);
  });
  it('should console.error if the variant is defined', () => {
    render(<Badge>completed</Badge>);

    expect(console.error).toHaveBeenCalled();
    expect(errorMessage).toMatch(/.*variant.*required.*/i);
  });
  it('should console.error if the variant is invalid', () => {
    render(<Badge variant="invalid">Completed</Badge>);

    expect(console.error).toHaveBeenCalled();
    expect(errorMessage).toMatch(/invalid.*variant.*/i);
  });
  it('should render correctly if the variant is valid', () => {
    Object.keys(COLORS).forEach((color) => {
      const { container } = render(<Badge variant={color}>badge</Badge>);

      expect(container).toMatchSnapshot();

      cleanup();
    });
  });
});

it('should return true', () => {
  expect(true).toBe(true);
});
