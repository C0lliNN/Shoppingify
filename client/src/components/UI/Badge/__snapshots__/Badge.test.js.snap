// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge/> should render correctly if the variant is valid 1`] = `
.c0 {
  border: 1px solid #F9A109;
  border-radius: 12px;
  color: #F9A109;
  padding: 3px 10px;
  font-weight: 500;
  font-size: 12px;
}

<div>
  <span
    class="c0"
  >
    badge
  </span>
</div>
`;

exports[`<Badge/> should render correctly if the variant is valid 2`] = `
.c0 {
  border: 1px solid #56CCF2;
  border-radius: 12px;
  color: #56CCF2;
  padding: 3px 10px;
  font-weight: 500;
  font-size: 12px;
}

<div>
  <span
    class="c0"
  >
    badge
  </span>
</div>
`;

exports[`<Badge/> should render correctly if the variant is valid 3`] = `
.c0 {
  border: 1px solid #EB5757;
  border-radius: 12px;
  color: #EB5757;
  padding: 3px 10px;
  font-weight: 500;
  font-size: 12px;
}

<div>
  <span
    class="c0"
  >
    badge
  </span>
</div>
`;

exports[`<Badge/> should render correctly if the variant is valid 4`] = `
.c0 {
  border: 1px solid #FFFFFF;
  border-radius: 12px;
  color: #FFFFFF;
  padding: 3px 10px;
  font-weight: 500;
  font-size: 12px;
}

<div>
  <span
    class="c0"
  >
    badge
  </span>
</div>
`;

exports[`<Badge/> should render correctly if the variant is valid 5`] = `
.c0 {
  border: 1px solid #E0E0E0;
  border-radius: 12px;
  color: #E0E0E0;
  padding: 3px 10px;
  font-weight: 500;
  font-size: 12px;
}

<div>
  <span
    class="c0"
  >
    badge
  </span>
</div>
`;

exports[`<Badge/> should render correctly if the variant is valid 6`] = `
.c0 {
  border: 1px solid #BDBDBD;
  border-radius: 12px;
  color: #BDBDBD;
  padding: 3px 10px;
  font-weight: 500;
  font-size: 12px;
}

<div>
  <span
    class="c0"
  >
    badge
  </span>
</div>
`;

exports[`<Badge/> should render correctly if the variant is valid 7`] = `
.c0 {
  border: 1px solid #828282;
  border-radius: 12px;
  color: #828282;
  padding: 3px 10px;
  font-weight: 500;
  font-size: 12px;
}

<div>
  <span
    class="c0"
  >
    badge
  </span>
</div>
`;

exports[`<Badge/> should render correctly if the variant is valid 8`] = `
.c0 {
  border: 1px solid #000000;
  border-radius: 12px;
  color: #000000;
  padding: 3px 10px;
  font-weight: 500;
  font-size: 12px;
}

<div>
  <span
    class="c0"
  >
    badge
  </span>
</div>
`;

exports[`<Badge/> should render correctly if the variant is valid 9`] = `
.c0 {
  border: 1px solid #34333A;
  border-radius: 12px;
  color: #34333A;
  padding: 3px 10px;
  font-weight: 500;
  font-size: 12px;
}

<div>
  <span
    class="c0"
  >
    badge
  </span>
</div>
`;

exports[`<Badge/> should render correctly if the variant is valid 10`] = `
.c0 {
  border: 1px solid #80485b;
  border-radius: 12px;
  color: #80485b;
  padding: 3px 10px;
  font-weight: 500;
  font-size: 12px;
}

<div>
  <span
    class="c0"
  >
    badge
  </span>
</div>
`;

exports[`<Badge/> should render correctly if the variant is valid 11`] = `
.c0 {
  border: 1px solid #fff0de;
  border-radius: 12px;
  color: #fff0de;
  padding: 3px 10px;
  font-weight: 500;
  font-size: 12px;
}

<div>
  <span
    class="c0"
  >
    badge
  </span>
</div>
`;
